src/lib/decorations/path.typ

```diff
@@ -59,6 +63,24 @@
   factor: 150%,
 )
 
+// Square default style
+#let square-default-style = (
+  ..default-style,
+  /// Midpoint factor
+  factor: 50%,
+)
+
+#let resolve-amplitude(amplitude, segment, num-segments) = {
+  segment = calc.max(0, calc.min(segment, num-segments))
+  return if type(amplitude) == function {
+    (amplitude)(segment / num-segments * 100%)
+  } else if type(amplitude) == array {
+    amplitude.at(calc.rem(int(segment), amplitude.len()), default: 0)
```

Comment:
Have you considered `amplitude.at(calc.rem(int(2*segment), amplitude.len()), default: 0)`? That gives the user control over individual peaks and troughs. With `int(segment)`, you can only control them in full-wavelength pairs.